.github/workflows/ci.yml

```diff
@@ -13,29 +13,58 @@ jobs:
     steps:
       - name: Clone This Repo
         uses: actions/checkout@v2
-      - name: Build
+      - name: Build with Sanitizers
         run: |
-          sudo apt-get install -y lcov
-          cmake -S test -B build/ \
+          sudo apt-get install -y cmake lcov
+          CFLAGS="  --coverage -O0 -Wall -Wextra"
+          CFLAGS+=" -Werror -Wno-error=pedantic"
+          CFLAGS+=" -D_FORTIFY_SOURCE=2"
+          CFLAGS+=" -Wformat"
+          CLFAGS+=" -Wformat-security"
+          CFLAGS+=" -Warray-bounds"
+          CFLAGS+=" -fsanitize=address,undefined"
+          CFLAGS+=" -fsanitize=pointer-compare -fsanitize=pointer-subtract"
+          CFLAGS+=" -fsanitize-recover=undefined"
+          CFLAGS+=" -fsanitize-address-use-after-scope"
+          CFLAGS+=" -fsanitize-undefined-trap-on-error"
+          CFLAGS_=" -fstack-protector-all"
+          cmake -S test -B build \
           -G "Unix Makefiles" \
           -DCMAKE_BUILD_TYPE=Debug \
-          -DCMAKE_C_FLAGS='--coverage -Wall -Wextra -DNDEBUG'
-          make -C build/ all
-      - name: Test
+          -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON \
+          -DCMAKE_C_FLAGS="${CFLAGS}"
+          make -C build all
+      - name: Run Tests with Sanitizers
         run: |
-          cd build/
+          cd build
+          make coverage
           ctest -E system --output-on-failure
           cd ..
-      - name: Run Coverage
+      - name: Build Tests for Coverage
         run: |
-          make -C build/ coverage
+          CFLAGS="  --coverage -O0 -Wall -Wextra"
+          CFLAGS+=" -Werror -Wno-error=pedantic"
+          CFLAGS+=" -DFORTIFY_SOURCE=0"
+          CFLAGS+=" -DNDEBUG"
```

Comment:
I don't feel great about using DNDEBUG to remove assert calls here, since technically assert calls are a decision point when enabled.

Reply:
The result of removing this is much lower branch / decision coverage.

Reply:
Enabling asserts in coverage builds requires testing internal functions that use asserts. We have an [example of including assert checks in unit testing](https://github.com/FreeRTOS/coreJSON/blob/main/test/unit-test/core_json_utest.c#L1706-L1838) within the coreJSON library.  The mechanism of catching asserts is defined in [catch_assert.h](https://github.com/FreeRTOS/coreJSON/blob/main/test/unit-test/catch_assert.h). 

We can follow the same mechanism of including asserts in unit tests for new development.